task/power/src/main.rs

```diff
@@ -349,6 +359,45 @@ macro_rules! max5970_controller {
     };
 }
 
+fn trace_max5970(dev: &Max5970, sensor: SensorId) {
+    if let Ok(Volts(volts)) = dev.max_vout() {
+        use drv_i2c_devices::max5970::Register;
+
+        if volts < 12.0 {
+            return;
+        }
+
+        ringbuf_entry!(Trace::Max5970 {
+            sensor,
+            status0: match dev.read_reg(Register::status0) {
+                Ok(reg) => reg,
+                _ => return,
+            },
+            status3: match dev.read_reg(Register::status3) {
+                Ok(reg) => reg,
+                _ => return,
+            },
+            fault0: match dev.read_reg(Register::fault0) {
+                Ok(reg) => reg,
+                _ => return,
+            },
+            fault1: match dev.read_reg(Register::fault1) {
+                Ok(reg) => reg,
+                _ => return,
+            },
+            fault2: match dev.read_reg(Register::fault2) {
+                Ok(reg) => reg,
+                _ => return,
+            },
+            max_current: match dev.max_iout() {
+                Ok(Amperes(amps)) => amps,
+                _ => return,
+            },
+            max_voltage: volts,
+        });
```

Comment:
I'd vote to explicitly log errors here, since they could be interesting.  Something like
```suggestion
        // Helper function to read a register or log the failure
        let f = |reg| match dev.read_reg(reg) {
            Ok(reg) => Some(reg),
            Err(err) => {
                ringbuf_entry!(Trace::Max5970ReadFailed { reg, err });
                None
            }
        };
        let Some(status0) = f(Register::status0) else { return; };
        let Some(status3) = f(Register::status3) else { return; };
        let Some(fault0) = f(Register::fault0) else { return; };
        let Some(fault1) = f(Register::fault1) else { return; };
        let Some(fault2) = f(Register::fault2) else { return; };

        let max_current = match dev.max_iout() {
            Ok(Amperes(amps)) => amps,
            Err(err) => {
                ringbuf_entry!(Trace::Max5970MaxCurrentFailed { err });
                return;
            }
        };

        ringbuf_entry!(Trace::Max5970 {
            sensor,
            status0,
            status3,
            fault0,
            fault1,
            fault2,
            max_current,
            max_voltage: volts,
        });
```

along with two new ringbuf variants:
```diff
diff --git a/task/power/src/main.rs b/task/power/src/main.rs
index cd5b0623..93a77268 100644
--- a/task/power/src/main.rs
+++ b/task/power/src/main.rs
@@ -48,6 +48,13 @@ enum Trace {
         max_current: f32,
         max_voltage: f32,
     },
+    Max5970ReadFailed {
+        reg: drv_i2c_devices::max5970::Register,
+        err: ResponseCode,
+    },
+    Max5970MaxCurrentFailed {
+        err: ResponseCode,
+    },
     None,
 }
```

Reply:
I have mixed feelings about this because I don't want to flood the ring buffer with a failed device and miss the PGs on others. (We will be able to deduce failures by the absence of expected devices.)